Add new API subscription version

MichalKalke opened the pull request:
<!--   Thank you for your contribution. Before you submit the pull request:
1. Follow contributing guidelines, templates, the recommended Git workflow, and any related documentation.
2. Read and submit the required Contributor Licence Agreements (https://github.com/kyma-project/community/blob/main/CONTRIBUTING.md#agreements-and-licenses).
3. Test your changes and attach their results to the pull request.
4. Update the relevant documentation.
-->

**Description**
Update `apply`, `init` and `sync` commands to be compatible with `hydroform main` and new subscription API with versioning.  

Changes proposed in this pull request:

- update `apply/function` to set `schemaVersion` to default when the version is not provided
- update `init/function` to pass the `SchemaVersion` to the config
- add in `sync/function` the flag to set the `schemaVersion`
- add in `kyma_sync_function.md` description of `--schemaVersion` flag

**Related issue(s)**
https://github.com/kyma-project/cli/issues/1532
[hydroform](https://github.com/kyma-project/hydroform/pull/397)

MichalKalke commented:
/test all

kwiatekus reviewed (changes requested):
I'd rather see this flag not using camel case format but 
`--schema-version`

I cant use the flag when calling kyma init commang
```bash
../bin/kyma-darwin init function --schemaVersion v1                                                                                        
Error: unknown flag: --schemaVersion
```

I also dont see it when calling `kyma init function --help`

MichalKalke commented:
> I'd rather see this flag not using camel case format but `--schema-version`
> 
> I cant use the flag when calling kyma init commang
> 
> ```shell
> ../bin/kyma-darwin init function --schemaVersion v1                                                                                        
> Error: unknown flag: --schemaVersion
> ```
> 
> I also dont see it when calling `kyma init function --help`

I have changed the flag to `schema-version` as you suggested.
I have also added above mentioned flag to the `kyma init function`.

MichalKalke commented:
/test all

MichalKalke commented:
/retest
